gensim/corpora/indexedcorpus.py

```diff
@@ -113,10 +121,12 @@ def __getitem__(self, docno):
         if self.index is None:
             raise RuntimeError("cannot call corpus[docid] without an index")
 
-        if isinstance(docno, slice):
+        if isinstance(docno, (slice, list, numpy.ndarray)):
             return utils.SlicedCorpus(self, docno)
-
-        return self.docbyoffset(self.index[docno])
+        elif isinstance(docno, (int, numpy.int)):
```

Comment:
`numpy.int` and `int` are the same thing; perhaps you meant `numpy.integer`? (to also cover int32, uint16 etc etc).